Updated Workspace Views article (#7944)
umbraco/UmbracoDocs: 16/umbraco-cms/customizing/extending-overview/extension-types/workspaces/workspace-views.md

# Workspace Views

> If you have worked with Umbraco versions prior to v14, you likely know this feature as Content Apps. In the new Management API and Web Component-based backoffice, this concept is unified under Workspaces.

Reviewer: I think this should still be within a note, right? It's historic data, and I think we should eventually take these out.

```suggestion
If you have worked with Umbraco versions prior to version 14, you likely know this feature as Content Apps. In the new Management API and Web Component-based backoffice, this concept is unified under Workspaces.
```

## Purpose

Workspace Views provide:

- **Tab-based organization** for different editing aspects.
- **Contextual interfaces** related to the current entity.
- **Workspace integration** with access to workspace contexts.
- **Custom functionality** specific to entity types.

<figure><img src="../../../../.gitbook/assets/workspace-views.svg" alt=""><figcaption><p>Workspace Views</p></figcaption></figure>

## Manifest

{% code caption="manifest.ts" %}

```typescript
{
  type: 'workspaceView',
  name: 'Example Counter Workspace View',
  alias: 'example.workspaceView.counter',
  element: () => import('./counter-workspace-view.js'),
  weight: 900,
  meta: {
	label: 'Counter',
	pathname: 'counter',
	icon: 'icon-lab',
  },
  conditions: [
	{
	  alias: UMB_WORKSPACE_CONDITION_ALIAS,
	  match: 'Umb.Workspace.Document',
	},
  ],
}
```

{% endcode %}

### Key Properties

- `weight` - Tab ordering (higher weight appears first)
- `meta.label` - Text displayed on the tab
- `meta.pathname` - URL segment for the view
- `meta.icon` - Icon displayed on the tab

> - `conditions` - Determines workspace availability

Reviewer: ```suggestion
- `weight` - Tab ordering (higher weight appears first).
- `meta.label` - Text displayed on the tab.
- `meta.pathname` - URL segment for the view.
- `meta.icon` - Icon displayed on the tab.
- `conditions` - Determines workspace availability.
```

  @state()
  private count = 0;

  constructor() {
	super();
	this.consumeContext(EXAMPLE_COUNTER_CONTEXT, (instance) => {
	  this.#counterContext = instance;
	  this.#observeCounter();
	});
  }

  #observeCounter(): void {
	if (!this.#counterContext) return;
	this.observe(this.#counterContext.counter, (count) => {
	  this.count = count;
	});
  }

  override render() {
	return html`
	  <uui-box class="uui-text">
		<h1 class="uui-h2">Counter Example</h1>
		<p class="uui-lead">Current count value: ${this.count}</p>
		<p>This workspace view consumes the Counter Context and displays the current count.</p>
	  </uui-box>
	`;
  }

  static override styles = [
	UmbTextStyles,
	css`
	  :host {
		display: block;
		padding: var(--uui-size-layout-1);
	  }
	`,
  ];
}

export default ExampleCounterWorkspaceView;

declare global {
  interface HTMLElementTagNameMap {
	'example-counter-workspace-view': ExampleCounterWorkspaceView;
  }
}
```

{% endcode %}

## Conditions

Conditions control when a Workspace View is shown. Every condition entry has an `alias` that references a built-in or custom condition, and a `match` value that the condition evaluates against.

{% hint style="info" %}

> UMB_WORKSPACE_CONDITION_ALIAS is a typed constant for Umb.Condition.WorkspaceAlias. Import it from `@umbraco-cms/backoffice/workspace` to get type safety when referencing workspace aliases in your manifest.

Reviewer: ```suggestion
`UMB_WORKSPACE_CONDITION_ALIAS` is a typed constant for `Umb.Condition.WorkspaceAlias`. Import it from `@umbraco-cms/backoffice/workspace` to get type safety when referencing workspace aliases in your manifest.
```
